### PR TITLE
move nicsdru_workflow module into origins repo 

### PR DESCRIPTION
**Pull request by DuttonMa**
module now renamed to origins_workflow

### CODE REVIEW

File: origins_workflow/src/Controller/AuditController.php [hunk 1]
```diff
@@ -0,0 +1,139 @@
...
+          '#suffix' => "</div>",
+          '#weight' => 0,
+        ];
+        // Build a confirm link.
+        $link_object = Link::createFromRoute($this->t($audit_button_text),
+          'origins_workflow.audit_controller_confirm_audit',
+          ['nid' => $nid],
+          ['attributes' => ['rel' => 'nofollow', 'class' => 'audit_link']]);
+        // Add confirm link to render array.
+        $render_array['link1'] = array_merge($link_object->toRenderable(),
```
Comment (omahm) [inline origins_workflow/src/Controller/AuditController.php]: Why not use the link render element?

File: origins_workflow/src/Controller/AuditController.php [hunk 2]
```diff
@@ -0,0 +1,139 @@
...
+   * Confirm Audit.
+   *
+   * @return string
+   *   Return confirmation string.
+   */
+  public function confirmAudit($nid) {
+    // Bump up the 'next audit due' date and log it.
+    $node = $this->entityTypeManager()->getStorage('node')->load($nid);
+    if ($node) {
+      $node->set('field_next_audit_due', date('Y-m-d', strtotime("+6 months")));
```
Comment (omahm) [inline origins_workflow/src/Controller/AuditController.php]: Will this always be 6 months?

File: origins_workflow/src/Controller/ModerationStateController.php [hunk 3]
```diff
@@ -0,0 +1,72 @@
...
+   */
+  public function changeState($nid, $new_state) {
+    // Load the entity.
+    $entity = $this->entityTypeManager->getStorage('node')->load($nid);
+    if ($entity) {
+      // Request the state change.
+      $entity->set('moderation_state', $new_state);
+      $entity->save();
+      // Log it.
+      $message = t('State of') . " '" . t($entity->getTitle()) . "' (nid " . $nid . ") " . t('changed to');
```
Comment (omahm) [inline origins_workflow/src/Controller/ModerationStateController.php]: No biggy but I'd look at using placeholders to keep this tidy.

File: origins_workflow/src/Form/AuditSettingsForm.php [hunk 4]
```diff
@@ -0,0 +1,224 @@
...
+      ->set('audit_content_types', $form_state->getValue('audit_content_types'))
+      ->save();
+  }
+
+  /**
+   * Remove audit field from the content type.
+   */
+  private function removeAuditField($type) {
+    // Remove audit field from this content type.
+    $field = FieldConfig::loadByName('node', $type, 'field_next_audit_due');
```
Comment (omahm) [inline origins_workflow/src/Form/AuditSettingsForm.php]: Should be able to access this from EntityConfig manager and inject that as a dependency.

File: origins_workflow/src/Form/AuditSettingsForm.php [hunk 5]
```diff
@@ -0,0 +1,224 @@
...
+    if (!empty($field)) {
+      // See if there is any data in this field.
+      $ids = \Drupal::entityQuery('node')
```
Comment (omahm) [inline origins_workflow/src/Form/AuditSettingsForm.php]: Use dependency injection.

File: origins_workflow/composer.json [hunk 6]
```diff
@@ -0,0 +1,14 @@
+{
+  "name": "dof-dss/origins_workflow",
+  "description": "Drupal module to provide workflow functionality for NICS websites",
+  "type": "drupal-module",
+  "authors": [
+    {
+      "name": "Martin Dutton",
+      "email": "mail@martinduttonconsulting.co.uk"
```
Comment (johangant) [inline origins_workflow/composer.json]: Not a point to hold this up on, but perhaps we should use a generic NICS address for this?

Comment (omahm) [reply origins_workflow/composer.json]: I think in future we should just rip out the addresses altogether

Comment (johangant) [reply origins_workflow/composer.json]: Nice. I'll create a new ticket to cover that across all repos

File: origins_workflow/src/Controller/AuditController.php (hunk 1 above)
Comment (DuttonMa) [reply origins_workflow/src/Controller/AuditController.php]: Not sure what you mean, please could you explain ?

Comment (omahm) [reply origins_workflow/src/Controller/AuditController.php]: Sure, if you use https://api.drupal.org/api/drupal/core%21lib%21Drupal%21Core%21Render%21Element%21Link.php/class/Link/8.7.x you don't need to create 2 separate elements and merge the arrays.

File: origins_workflow/src/Controller/AuditController.php (hunk 2 above)
Comment (DuttonMa) [reply origins_workflow/src/Controller/AuditController.php]: Good idea, but have added TODO and will implement later

**Pull request merged**